### ISSUE TITLE
[Bug] Targeting feature in phases has no effect

### ISSUE THREAD

**Issue by bosyk** [growthbook/growthbook#1383]
### Summary

In experiments, the targeting feature in phases has no effect on the analysis result:

<img width="300" alt="image" src="https://github.com/growthbook/growthbook/assets/30110352/7a1afe4b-07d7-410d-b696-ebb015dcc142">

#### Expected Behavior

Setting a traget audience in a phase does filter the experiment participants according to that filter when performing an analysis in for that phase.

#### Current Behavior

Setting a traget audience in a phase does not filter the experiment participants according to that filter when performing an analysis.

#### Useful information

I verified this behaviour with internal and the following synthetic data:
[gb-experiment-viewed-phase-bug.csv.gz](https://github.com/growthbook/growthbook/files/11832963/gb-experiment-viewed-phase-bug.csv.gz)
[gb-metric-phase-bug.csv.gz](https://github.com/growthbook/growthbook/files/11832964/gb-metric-phase-bug.csv.gz)

with the following configuration

```yaml
    secureAttributeSalt: ''
datasources:
  ds_405wf81mlj4gdhbt:
    type: postgres
    name: postgres
    params:
      host: postgres
      port: 5432
      database: postgres
      user: postgres
      password: ''
    settings:
      schemaFormat: custom
      userIdTypes:
        - userIdType: user_id
          description: Logged-in user id
      queries:
        exposure:
          - id: user_id
            userIdType: user_id
            dimensions:
              - country
            name: Logged-in Users
            description: ''
            query: |-
              SELECT
                user_id as user_id,
                timestamp as timestamp,
                experiment_id as experiment_id,
                variation_id as variation_id,
                country as country
              FROM
                experiments_viewed
            hasNameCol: false
        identityJoins: []
      events:
        experimentEvent: $experiment_started
        experimentIdProperty: Experiment name
        variationIdProperty: Variant name
      informationSchemaId: inf_405wf81mlj4gdhfp
metrics:
  met_405wf81mlj4gemtz:
    type: binomial
    name: Metric
    datasource: ds_405wf81mlj4gdhbt
    conversionWindowHours: 72
    loseRisk: 0.0125
    winRisk: 0.0025
    maxPercentChange: 0.5
    minPercentChange: 0.005
    minSampleSize: 150
    userIdTypes:
      - user_id
    sql: |-
      SELECT
        user_id as user_id,
        timestamp as timestamp
      FROM
        metric
    queryFormat: sql
    userIdColumns:
      user_id: user_id
      anonymous_id: anonymous_id
```

I discovered the experiment "1" from the data source and evaluated it with the metric "Metric". Target in the phase was determined by filtering on dimension "country".

**Comment by lukesonnet** [comment]
Hi, currently the experience of changing settings like this in the Experiment page is not quite what most users expect. For now it largely just affects meta-data and only changes to the feature itself actually change targeting. We're working on this.

But it seems like in your case you don't want to change targeting, just the analysis. In that case, I suggest you use Dimensions or Segments.

**Comment by lukesonnet** [comment]
New experiments with Linked Feature Flags (available as of release [2.4](https://github.com/growthbook/growthbook/releases/tag/v2.4.0)) will automatically update the targeting of associated feature flags when you change the targeting from the experiment page.
